backend/src/recommitment/recommitment.service.ts

```diff
@@ -62,56 +64,91 @@ export class RecommitmentService {
       recommitmentCycle.reinitiationEndDate
         ? recommitmentCycle.reinitiationEndDate
         : recommitmentCycle.endDate;
-    if (isAfter(currentPST, addDays(endDate, 1)) || isAfter(recommitmentCycle.startDate, currentPST)) {
-      throw new ForbiddenException('Unable to update recommitment status outside of recommitment cycle end date.');
+    if (
+      isAfter(currentPST, addDays(endDate, 1)) ||
+      isAfter(recommitmentCycle.startDate, currentPST)
+    ) {
+      throw new ForbiddenException(
+        'Unable to update recommitment status outside of recommitment cycle end date.',
+      );
     }
-    await this.recommitmentRepository.update(
-      {
+    const existingRecommitment = await this.recommitmentRepository.findOne({
+      where: {
         personnelId: id,
         recommitmentCycleId: new Date().getFullYear(),
         program: key,
       },
-      {
-        status: recommitmentUpdate[key]?.status,
-        memberDecisionDate: null,
-        memberReason: null,
-        supervisorIdir: null,
-        supervisorReason: null,
-      },
-    );
-
-    const recommitment = await this.recommitmentRepository.findOneOrFail({ where: {
-      personnelId: id,
-      recommitmentCycleId: new Date().getFullYear(),
-      program: key,
-    }, relations: ['personnel', 'recommitmentCycle'] });
-
-    const otherProgram = key === Program.BCWS ? Program.EMCR : Program.BCWS;
-    const personnel = await this.personnelService.findOneById(id);
-    const otherProgramPersonnel = otherProgram === Program.BCWS ?
-      personnel.bcws :
-      personnel.emcr;
-    let otherProgramStatusReset = false;
-    if (otherProgramPersonnel) {
-      const recommitmentOtherProgram = await this.recommitmentRepository.findOne({
-        where: {
+    });
```

Comment:
Is this going to affect recommitment as well as reinitiation?

Reply:
This should only happen when you click "Restart Recommitment" as a coordinator, which i think only happens during the reinit period